src/magicgui/backends/_qtpy/widgets.py

```diff
@@ -952,9 +952,12 @@ def _mgui_set_choices(self, choices: Iterable[tuple[str, Any]]) -> None:
             for i in reversed(range(self._qwidget.count())):
                 if self._qwidget.itemText(i) not in choice_names:
                     self._qwidget.removeItem(i)
-            # update choices
+            # update choices and insert separators
             for name, data in choices_:
-                self._mgui_set_choice(name, data)
+                if data is Separator:
+                    self._qwidget.insertSeparator(self._mgui_get_count())
```

Comment:
also, this conditional should probably move one level lower to `_mgui_set_choice`